prow/plugins/milestoneapplier/milestoneapplier_test.go

```diff
@@ -0,0 +1,168 @@
+			prAction:            github.PullRequestActionClosed,
+			baseBranch:          "master",
+			merged:              true,
+			previousMilestone:   0,
+			configuredMilestone: 2,
+			expectedMilestone:   2,
+		},
+		{
+			name:                "opened PR on non-default branch => add milestone",
+			baseBranch:          "release-1.0",
+			prAction:            github.PullRequestActionOpened,
+			configuredMilestone: 1,
+			expectedMilestone:   1,
+		},
+		{
+			name:                "synced PR on non-default branch => do nothing",
+			baseBranch:          "release-1.0",
+			prAction:            github.PullRequestActionSynchronize,
+			previousMilestone:   0,
+			configuredMilestone: 1,
+			expectedMilestone:   0,
+		},
+		{
+			name:              "closed (not merged) PR on non-default branch => do nothing",
+			baseBranch:        "release-1.0",
+			prAction:          github.PullRequestActionClosed,
+			merged:            false,
+			expectedMilestone: 0,
+		},
+		{
+			name:                "merged PR but has existing milestone on non-default branch => do nothing",
+			baseBranch:          "release-1.0",
+			prAction:            github.PullRequestActionClosed,
+			merged:              true,
+			previousMilestone:   2,
+			configuredMilestone: 1,
+			expectedMilestone:   2,
+		},
+		{
+			name:                "merged PR but does not have existing milestone on non-default branch => add milestone",
+			prAction:            github.PullRequestActionClosed,
+			baseBranch:          "release-1.0",
+			merged:              true,
+			previousMilestone:   0,
+			configuredMilestone: 1,
+			expectedMilestone:   1,
+		},
+	}
+
+	for _, tc := range testcases {
+		basicPR := github.PullRequest{
+			Number: 1,
+			Base: github.PullRequestBranch{
+				Repo: github.Repo{
+					Owner: github.User{
+						Login: "kubernetes",
+					},
+					Name:          "kubernetes",
+					DefaultBranch: "master",
+				},
+				Ref: tc.baseBranch,
+			},
+		}
+
+		sha := "abcdef"
+		basicPR.Merged = tc.merged
+		if tc.merged {
+			basicPR.MergeSHA = &sha
+		}
+		if tc.previousMilestone != 0 {
+			basicPR.Milestone = &github.Milestone{
+				Number: tc.previousMilestone,
+			}
+		}
+
+		event := github.PullRequestEvent{
+			Action:      tc.prAction,
+			Number:      basicPR.Number,
+			PullRequest: basicPR,
+		}
+
+		fakeClient := &fakegithub.FakeClient{
+			PullRequests: map[int]*github.PullRequest{
+				basicPR.Number: &basicPR,
+			},
+			MilestoneMap: milestonesMap,
+			Milestone:    tc.previousMilestone,
+		}
+
+		var configuredMilestoneTitle string
+		for title, number := range milestonesMap {
+			if number == tc.configuredMilestone {
+				configuredMilestoneTitle = title
+			}
+		}
+
+		if err := handle(fakeClient, logrus.WithField("plugin", pluginName), configuredMilestoneTitle, event); err != nil {
+			t.Errorf("(%s): Unexpected error from handle: %v.", tc.name, err)
+			continue
```

Comment:
nit: it seems clearer if either t.Fatalf was used, or the if after this was an "else if" instead.

note: if you use t.Fatalf, then the rest of the tests won't run. One way to circumvent that is to use t.Run to run each case as a subtest. Some examples in [lgtm](https://github.com/kubernetes/test-infra/blob/ea4f8aa6121e16f4c87bd570be2dc629a0828360/prow/plugins/lgtm/lgtm_test.go#L929-L930) and [retitle](https://github.com/kubernetes/test-infra/blob/ea4f8aa6121e16f4c87bd570be2dc629a0828360/prow/plugins/retitle/retitle_test.go#L133-L134) plugins